src/runtime/composables/useDirectusToken.ts

```diff
@@ -1,14 +1,86 @@
 import { useCookie, useNuxtApp, useRuntimeConfig } from "#app";
+import { CookieRef } from "~~/node_modules/nuxt3/dist/app/composables/cookie";
+import { DirectusAuthResponse } from "../types";
+
 
 export const useDirectusToken = () => {
   const nuxtApp = useNuxtApp();
+  const baseURL = useDirectusUrl();
+  const config = useRuntimeConfig()
+
+  const accessToken = (): CookieRef<string> => {
+    nuxtApp._cookies = nuxtApp._cookies || {};
+    if (nuxtApp._cookies.directus_token) {
+      return nuxtApp._cookies.directus_token;
+    }
+
+    const cookie = useCookie<string | null>("directus_token", config.directus.cookie);
+    nuxtApp._cookies.directus_token = cookie;
+    return cookie;
+  };
+
+  const refreshToken = (): CookieRef<string> => {
+    nuxtApp._cookies = nuxtApp._cookies || {};
+    if (nuxtApp._cookies.directus_refresh_token) {
+      return nuxtApp._cookies.directus_refresh_token;
+    }
+
+    const cookie = useCookie<string | null>("directus_refresh_token", config.directus.cookie);
+    nuxtApp._cookies.directus_refresh_token = cookie;
+    return cookie;
+  };
+
+  const expiredAt = (): CookieRef<number> => {
+    nuxtApp._cookies = nuxtApp._cookies || {};
+    if (nuxtApp._cookies.directus_token_expired_at) {
+      return nuxtApp._cookies.directus_token_expired_at;
+    }
+
+    const cookie = useCookie<number | null>("directus_token_expired_at", config.directus.cookie);
+    nuxtApp._cookies.directus_token_expired_at = cookie;
+    return cookie;
+  };
+
+  const refreshTokens = async (): Promise<DirectusAuthResponse | null> => {
+    if (refreshToken() && refreshToken().value) {
+      const body = {
+        refresh_token: refreshToken().value,
+      };
+      const data = await $fetch<{data: DirectusAuthResponse}>("/auth/refresh", {
+        baseURL,
+        body,
+        method: "POST"
+      });
+      expiredAt().value = new Date().getTime() + data.data.expires;
+      accessToken().value = data.data.access_token;
+      refreshToken().value = data.data.refresh_token;
+      return data.data;
+    } else {
+      return null;
+    }
+  };
 
-  nuxtApp._cookies = nuxtApp._cookies || {};
-  if (nuxtApp._cookies.directus_token) {
-    return nuxtApp._cookies.directus_token;
-  }
+  const token = async (): Promise<string | null> => {
+    if (expiredAt() && expiredAt().value) {
+      const isExpired = new Date().getTime() > expiredAt().value;
+      if (isExpired && refreshToken() && refreshToken().value) {
```

Comment:
if token is expired I refresh the tokens

Reply:
sorry, i actually miss that =D 

So yes, it dose exactly how the SDK dose.